src/compose/ComposeBox.js

```diff
@@ -320,15 +315,11 @@ class ComposeBox extends PureComponent<Props, State> {
   };
 
   handleSend = () => {
-    const { dispatch, narrow, caughtUp, _ } = this.props;
+    const { dispatch, _ } = this.props;
     const { message } = this.state;
+    const narrow = this.getDestinationNarrow();
 
-    if (!caughtUp.newer) {
-      showErrorAlert(_('Failed to send message'));
-      return;
-    }
```

Comment:
Will this reintroduce #3800? If not, why not?

In any case I don't think this benefits from being rolled into a long cleanup series like this one -- it's entirely about a behavior change, not anything to do with cleaning up the code. Better to have its own PR thread, to avoid adding complexity to this one.

Reply:
This does — I hadn't realized this history of this. I had thought that something about this PR made the symptoms described in [#mobile > Failed to send on Android](https://chat.zulip.org/#narrow/stream/48-mobile/topic/Failed.20to.20send.20on.20Android) worse, but looking at it again, I don't think that's the case.